src/js/services/profileService.js

```diff
@@ -324,6 +324,17 @@ angular.module('copayApp.services')
       });
     };
 
+    root.setMetaData = function(walletClient, addressBook, historyCache, cb) {
+
+      storageService.setAddressbook(walletClient.credentials.network, JSON.stringify(addressBook), function(err) {
```

Comment:
Since address book is global (per network type) It seems it would be better to 'merge' the backup's addressbook with the current one, (and removing duplicates).